exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/registry/FunctionRegistryHolder.java

```diff
@@ -97,9 +98,13 @@
   // function name, Map<function signature, function holder>
   private final Map<String, Map<String, DrillFuncHolder>> functions;
 
+  // jar name, List<function holder>
+  private final Map<String, List<FunctionHolder>> jarFunctions;
```

Comment:
I don't see any value of adding one more holder to keep functions. Current two holders have enough information and its unique. Your holder just duplicates it. Please consider approach i have suggested before.

Reply:
The `functions` holder groups functionHolders from different sources as long as they share the same name. As a result, for a given function name, the source of a specific signature cannot be inferred. Since the holders non-primitive datatypes, they are not copied and only references maintained. So their additional footprint should be fairly small within the Drillbit.